src/ivalue_manager.rs

```diff
@@ -287,9 +376,12 @@ impl<'a> WriteHolder<IValue, IValue> for IValueKeyHolderWrite<'a> {
     fn bool_toggle(&mut self, path: Vec<String>) -> Result<bool, RedisError> {
         let mut res = None;
         self.do_op(path, |v| {
-            let val = v.to_bool().unwrap() ^ true;
-            res = Some(val);
-            Ok(Some(IValue::from(val)))
+            if let DestructuredMut::Bool(mut bool_mut) = v.destructure_mut() {
```

Comment:
Can you explain what `DestructuredMut` is for?

Reply:
It is an `Enum` used to obtain a mutable reference to `IValue`,
Specifically for Boolean, `IValue` does not directly contain Boolean,
So this is a proxy type which allows getting and setting the original `IValue` as Boolean.

Reply:
Please document.